src/Text/Pandoc/Format.hs

```diff
@@ -0,0 +1,118 @@
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{- |
+   Module      : Text.Pandoc.Format
+   Copyright   : © 2022 Albert Krewinkel
+   License     : GPL-2.0-or-later
+   Maintainer  : Albert Krewinkel <pandoc@tarleb.com>
+
+Handling of format specifiers for input and output.
+-}
+module Text.Pandoc.Format
+  ( Spec (..)
+  , ExtensionsConfig (..)
+  , ExtensionsDiff (..)
+  , parseFormatSpec
+  , applyExtensionsDiff
+  , getExtensionsConfig
+  ) where
+
+import Control.Monad.Except (throwError)
+import Data.List (foldl')
+import Text.Pandoc.Class (PandocMonad)
+import Text.Pandoc.Error (PandocError (..))
+import Text.Pandoc.Extensions
+  ( Extension
+  , Extensions
+  , disableExtension
+  , enableExtension
+  , extensionEnabled
+  , getAllExtensions
+  , getDefaultExtensions
+  , readExtension
+  )
+import Text.Parsec
+import qualified Data.Text as T
+
+-- | Format specifier with the format's name and the lists of extensions
+-- to be enabled or disabled.
+data Spec = Spec
+  { specFormat     :: T.Text
+  , specExtsDiff   :: ExtensionsDiff
+  }
```

Comment:
In my mind, I was using `FlavoredFormat` for the format plus the *final* list of extensions, where this contains the *diff*. But I guess the name still fits, and FlavoredFormat would probably be a good name to use for this structure.

The switch from including the diff instead of the final extensions was made to [enable](https://github.com/jgm/pandoc/pull/8141/commits/854cf4c5ec39f72e6c9607d8f4fb19692e133512#diff-481541da9fda71a288df70ce119f38af46e15d65df5eaed763fa8b306b357142R109-R111) handling of extensions in custom readers/writers (#8141).

Reply:
OK, I see how you might want to say that this is like a plan for computing a format, rather than a format itself.  I don't have very strong feelings about the terminology, I just wanted to raise the question.  Keeping it as `Spec` is fine if you like that better.

Reply:
I think I prefer FlavoredFormat, so I changed it back. I'd be very happy to adjust names, I felt rather unsure about the ones here.